django/db/models/base.py

```diff
@@ -1169,6 +1194,15 @@ def _do_insert(self, manager, using, fields, returning_fields, raw):
             raw=raw,
         )
 
+    def _assign_returned_values(self, returned_values, returning_fields):
+        returning_fields_iter = iter(returning_fields)
+        for value, field in zip(returned_values, returning_fields_iter):
+            setattr(self, field.attname, value)
+        # Get rid of local values for all fields that were meant to be returned
+        # but weren't as they they are stale.
+        for field in returning_fields_iter:
+            self.__dict__.pop(field.attname, None)
```

Comment:
There might be better way to do that. The goal of this logic is to clear the local value as to mark the field as deferred so it's fetched from the database on access.